include/criteria.h

```diff
@@ -11,6 +11,14 @@
 struct mako_config;
 struct mako_notification;
 
+enum operator { OP_NONE, OP_EQUALS, OP_REGEX_MATCHES, OP_NOT_EQUALS, OP_TRUTHY, OP_FALSEY };
+
+struct mako_condition {
```

Comment:
This is specific to strings, maybe should be renamed to e.g. `struct mako_string_criteria`?

Reply:
I think I was originally thinking that mako_condition could be used for all of the conditions (and maybe there are other operators that other types of conditions could use, like "in" or "less than"/"greater than" for urgency, group-index).

Since this PR is "Generalize String criteria matching", I didn't want to go too far out of scope and refactor all of the conditions.

But I'm not too invested in this, so if you really want the rename, I will probably do it.

Reply:
How would that work? The `value` field is a `char *`, but we'd need some other type for urgency/group-index/etc.